CMakeLists.txt

```diff
@@ -1,504 +1,2030 @@
+  if (NOT (is_gcc OR is_clang))
+    message (FATAL_ERROR "san option only works for clang/gcc")
+  endif ()
+  if (san STREQUAL "memory" AND NOT (is_clang AND
+    CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 7))
+    message (FATAL_ERROR "memory sanitizer only available for clang 7+")
+  endif ()
+endif ()
+
+# the remaining options are obscure and rarely used
+option (beast_no_unit_test_inline
+  "Prevents unit test definitions from being inserted into global table"
+  OFF)
+option (beast_force_debug
+  "Force BEAST_DEBUG regardless of DEBUG settings"
+  OFF)
+# NOTE - THIS OPTION CURRENTLY DOES NOT COMPILE :
+# TODO: fix or remove
+option (verify_nodeobject_keys
+  "This verifies that the hash of node objects matches the payload. \
+  This check is expensive - use with caution."
+  OFF)
+option (dump_leaks_on_exit
+  "Displays heap blocks and counted objects which were not disposed of\
+  during exit. Only implemented for windows builds."
+  ON)
+option (single_io_service_thread
+  "Restricts the number of threads calling io_service::run to one. \
+  This can be useful when debugging."
+  OFF)
+option (boost_show_deprecated
+  "Allow boost to fail on deprecated usage. Only useful if you're trying\
+  to find deprecated calls."
+  OFF)
+# beast_check_mem_leaks can't be an option() because we want to support an
+# "undefined" which means use default behavior
+# option (beast_check_mem_leaks
+#   "Force beast mem leak checking. Default is on for DEBUG builds. Only implemented on WIN32"
+if (WIN32)
+  option (beast_disable_autolink "Disables autolinking of system libraries on WIN32" OFF)
+else ()
+  set (beast_disable_autolink OFF CACHE BOOL "WIN32 only" FORCE)
+endif ()
+if (coverage)
+  message (STATUS "coverage build requested - forcing Debug build")
+  set (CMAKE_BUILD_TYPE Debug)
+endif ()
+
+if (is_linux AND NOT static)
+  set (BUILD_SHARED_LIBS ON)
+else ()
+  if (BUILD_SHARED_LIBS)
+    message (WARNING "forcing BUILD_SHARED_LIBS OFF for this platform/config")
+  endif ()
+  set (BUILD_SHARED_LIBS OFF)
+endif ()
+
+#[===================================================================[
+   setup project-wide compiler settings
+#]===================================================================]
+
+#[=========================================================[
+   TODO some/most of these common settings belong in a
+   toolchain file, especially the ABI-impacting ones
+#]=========================================================]
+add_library (common INTERFACE)
+add_library (Ripple::common ALIAS common)
+# add a single global dependency on this interface lib
+link_libraries (Ripple::common)
+set_target_properties (common
+  PROPERTIES INTERFACE_POSITION_INDEPENDENT_CODE ON)
+target_compile_features (common INTERFACE cxx_std_14)
+target_compile_definitions (common
+  INTERFACE
+    $<$<CONFIG:Debug>:_DEBUG>
+    $<$<AND:$<BOOL:profile>,$<NOT:$<BOOL:assert>>>:NDEBUG>
+    OPENSSL_NO_SSL2         # << WHAT DOES THIS DO ?
```

Comment:
Still a TODO to look into this?

Reply:
it looks like it tries to disable SSLv2. This is also an option when configuring/building OpenSSL, but I guess if we do this it probably hides any v2 implementations (?). In any event, it belongs on the SSL lib, so I've added it as an interface compile def to the imported lib.

Reply:
It seems to hide the implementations of a given version number just by `#ifdef`. "SSLv2 is completely broken, and you should disable it during configuration." from (the OpenSSL wiki)[https://wiki.openssl.org/index.php/Compilation_and_Installation] doesn't make me too worried about this flag though 😆